src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs

```diff
@@ -341,5 +336,37 @@ private static IEnumerable<string> GetProjectJsonFilesInDirectory(string path)
                 }
             }
         }
+
+        /// <summary>
+        /// Returns a unique list of sources. New sources will be cached
+        /// and shared between restores.
+        /// </summary>
+        private static List<SourceRepository> GetSources(
+            CommandOption sources,
+            CommandOption fallBack,
+            ISettings settings)
+        {
+
+            // CommandLineSourceRepositoryProvider caches repositories to avoid duplicates
+            var packageSourceProvider = new PackageSourceProvider(settings);
+            var sourceProvider = new CommandLineSourceRepositoryProvider(packageSourceProvider);
+
+            // Take the passed in sources
+            var packageSources = sources.Values.Select(s => new PackageSource(s));
+
+            // If no sources were passed in use the NuGet.Config sources
+            if (!packageSources.Any())
+            {
+                // Add enabled sources
+                packageSources = packageSourceProvider.LoadPackageSources().Where(source => source.IsEnabled);
```

Comment:
I see this query is quite common. Consider adding an optional parameter to LoadPackageSources or an extension method to avoid duplication.